api-report/purchases-js.api.md

```diff
@@ -312,13 +312,13 @@ export class PurchasesError extends Error {
     constructor(
     errorCode: ErrorCode,
     message?: string,
-    underlyingErrorMessage?: string | null | undefined,
+    underlyingErrorMessage?: (string | null) | undefined,

```

Comment:
why are we doing this?

Reply:
This is the result of running prettier after installing it with yarn.

Not sure what changes, I'll double check how to fix it.

Reply:
Ok mystery solved, this was caused by the update of api-extractor.
I'll try to check if there's a way to configure this behaviour.

It's generated looking at the `purchase-operation-helper` `PurchaseFlowError` class.

Which has the following constructor
```
constructor(
    public readonly errorCode: PurchaseFlowErrorCode,
    message?: string,
    public readonly underlyingErrorMessage?: string | null,
    public readonly purchasesErrorCode?: ErrorCode,
    public readonly extra?: PurchasesErrorExtra,
  ) {
    super(message);
  }
```

The `(string|null)` is defined by the type and the `undefined` is added by the ?

Reply:
I didn't find a way to configure the api extractor.
We should change the api to `underlyingErrorMessage?: string` so that the type won't be `(string | null)`.

Since these files are generated and we have already a lot of changes I wouldn't do it as part of this email.